File: examples/example-kubernetes-client-openapi-operator-java/src/main/java/micronaut/operator/OnUpdateFilter.java
package micronaut.operator;

import io.micronaut.kubernetes.client.openapi.model.V1ConfigMap;
import jakarta.inject.Singleton;

import java.util.function.BiPredicate;

@Singleton
public class OnUpdateFilter implements BiPredicate<V1ConfigMap, V1ConfigMap> {
    @Override
    public boolean test(V1ConfigMap oldConfigMap, V1ConfigMap newConfigMap) {
        return newConfigMap.getMetadata().getAnnotations() != null
            && newConfigMap.getMetadata().getAnnotations().containsKey("io.micronaut.operator");
    }
}

File: examples/example-kubernetes-client-openapi-operator-python/src/main/python/micronaut/operator/OnUpdateFilter.py
# tag::class[]
from java.util.function import BiPredicate
from jakarta.inject import Singleton
from micronaut.kubernetes.client.openapi.model import V1ConfigMap
# end::class[]
from micronaut.context.annotation import Requires
from micronaut.context.env import Environment
# tag::class[]

# end::class[]

@Requires(env=Environment.KUBERNETES)
# tag::class[]
@Singleton
class OnUpdateFilter(BiPredicate[V1ConfigMap, V1ConfigMap]):
    def test(self, old_config_map: V1ConfigMap, new_config_map: V1ConfigMap) -> bool:
        annotations = new_config_map.getMetadata().getAnnotations()
        return annotations is not None and "io.micronaut.operator" in annotations
# end::class[]
